src/components/apphost.js

```diff
@@ -275,7 +275,7 @@ const supportedFeatures = function () {
      */
 function doExit() {
     try {
-        if (window.NativeShell) {
+        if (window.NativeShell && window.NativeShell.AppHost.exit) {
```

Comment:
I would suggest using optional chaining for this type of checks.

```suggestion
        if (window.NativeShell?.AppHost?.exit) {
```